File: fsm/data/serialize.m
input_filename = 'input.data';
output_filename = 'output.matlab';
output_stream = [];

file = fopen(input_filename, 'r');
data = fscanf(file, '%s');
fclose(file);

[l, r] = size(data);

for i = 1:r
    letter = data(i);
    val = fliplr(dec2bin(double(letter)));
    [a, width] = size(val);
    for j = (width + 1):8;
        val = [val '0'];
    end
    
    output_stream = [output_stream '01' val '00'];
end

file = fopen(output_filename, 'w');
fprintf(file, output_stream);
fclose(file);

File: fsm/data/serialize.py
input_filename = 'input.data';
output_filename = 'output.python';

stream = []
with open(input_filename, 'r') as f:
	line = f.readline()
	for c in line:
		if c == '\n':
			break
		value = int(ord(c))
		string = bin(value)[2:]
		string = string[::-1] + '0' * (8 - len(string))
		
		string = '01' + string + '00'
		stream.append(string)

with open(output_filename, 'w') as f:
	for i in stream:
		f.write("%s\n" % i)
